R/utils.R

```diff
@@ -83,3 +83,13 @@ open_manual_tests <- function() {
     utils::browseURL(svglite_manual_tests[[test]])
   })
 }
+
+checkIntFormat <- function(s) {
```

Comment:
I'd like to avoid copying code from base R as it limits which license we can use in the future. Is there any reason for this check in general? is there a failure with certain file names?

Reply:
I did some tests with `snprintf` and you get a segmentation fault if you use `"%s.svg"` (and perhaps with others, I didn't check all patterns). Something like `%f` does work but for the file `"one %f two %d .svg"` you get `one 0.000000 two 1 .svg`. That's not a crash but I also cannot imagine any use case for multiple patterns or for non-integer patterns.

> I'd like to avoid copying code from base R as it limits which license we can use in the future.

If you prefer I can also rewrite this check. I copied the code to ensure consistency with base R.

Reply:
I see — thanks for checking that. I’d prefer to have something rewritten rather than copied as it makes licensing easier

Reply:
Okay so I refactored `checkIntFormat` to `invalid_filename`. I'm not a legal expert on whether the function is sufficiently different, but at least it looks a bit different now.